### ISSUE TITLE
How to get this number 35?

### ISSUE THREAD

**Issue by Bruceluwill** [ywh187/FitPrune#6]
Thank you for your work!
I am confused about this line of code.

attn_self_attention = attn_weights_headmax[35:35+left_vtoken_num, 35:35 + left_vtoken_num] / left_vtoken_num 

I don't quite understand how the number 35 is used to obtain the image self-attention in this line of code, is it because the previous fixed value is for the text tokens? It might be due to my lack of knowledge, and I would appreciate it if you could take some time to explain it to me.

**Comment by ywh187** [comment]
Hello. The **35** in the code refers to the length of the **system prompt**. It's used to skip over the system prompt tokens and focus on the image tokens.

**Comment by Bruceluwill** [comment]
Thank you for your reply. Is the length of the system prompt a fixed value set by LLaVA 1.5, or did we set it ourselves?

**Comment by ywh187** [comment]
You are correct. Different MLLMs typically have system prompts of varying lengths.
